core/templates/dev/head/pages/exploration_editor/editor_tab/StateResponses.js

```diff
@@ -86,22 +86,23 @@ oppia.factory('responsesService', [
             .can_have_solution && stateSolutionService.savedMemento !== null) {
           if (stateSolutionService.savedMemento.correctAnswer !== null) {
             var currentStateName = editorContextService.getActiveStateName();
-            SolutionHelperService.verifySolution(
+            SolutionVerificationService.verifySolution(
               explorationContextService.getExplorationId(),
               explorationStatesService.getState(currentStateName),
               stateSolutionService.savedMemento.correctAnswer,
               function() {
                 explorationStatesService.updateSolutionValidity(
-                  currentStateName, false);
+                  currentStateName, true);
                 alertsService.addInfoMessage(
-                  'That solution does not lead to the next state!');
+                  'The solution is now valid!');
```

Comment:
A solution can be specified only after a response is added. So this message should show only when responses are modified after specifying a wrong solution.